app/index.js

```diff
@@ -107,6 +111,19 @@ var NodeModuleGenerator = yeoman.Base.extend({
                 type: 'input',
                 name: 'username',
                 message: 'What is your GitHub username?',
+                store: true,
+                validate: function(input) {
+                    var res =   !input.startsWith('_')   &&
```

Comment:
You're already using a regex, I meant to combine all the validation rules into a single regex. Performance is not at all a concern, since we run this only once on a single string.